static/app/views/insights/mobile/screens/components/screensOverview.tsx

```diff
@@ -0,0 +1,235 @@
+    SpanMetricsField.PROJECT_ID,
+    SpanMetricsField.TRANSACTION,
+    `count()`,
+    'avg(measurements.app_start_cold)',
+    'avg(measurements.app_start_warm)',
+    `avg(measurements.time_to_initial_display)`,
+    `avg(measurements.time_to_full_display)`,
+  ];
+
+  const spanMetricsDataset = DiscoverDatasets.SPANS_METRICS;
+  const spanMetricsFields = [
+    SpanMetricsField.PROJECT_ID,
+    SpanMetricsField.TRANSACTION,
+    `avg(mobile.slow_frames)`,
+    `avg(mobile.frozen_frames)`,
+    `avg(mobile.frames_delay)`,
+  ];
+
+  const isSpanPrimary = spanMetricsFields.some(
+    field => getAggregateAlias(field) === sortField
+  );
+  const primaryDataset = isSpanPrimary ? spanMetricsDataset : transactionMetricsDataset;
+  const primaryFields = isSpanPrimary ? spanMetricsFields : transactionMetricsFields;
+
+  const secondaryDataset = isSpanPrimary ? transactionMetricsDataset : spanMetricsDataset;
+  const secondaryFields = isSpanPrimary ? transactionMetricsFields : spanMetricsFields;
+
+  const {
+    data: primaryData,
+    isLoading: primaryLoading,
+    pageLinks: primaryLinks,
+    eventView: primaryEventView,
+  } = useMetrics(
+    primaryDataset,
+    primaryFields,
+    isProjectCrossPlatform ? selectedPlatform : undefined,
+    selection,
+    location,
+    sortedBy,
+    true,
+    []
+  );
+
+  const {
+    data: secondaryData,
+    // isLoading: secondaryLoading,
+    // pageLinks: secondaryLinks,
+    // eventView: secondaryEventView,
+  } = useMetrics(
+    secondaryDataset,
+    secondaryFields,
+    isProjectCrossPlatform ? selectedPlatform : undefined,
+    selection,
+    location,
+    undefined,
+    visibleScreens.length > 0,
+    visibleScreens
+  );
+
+  useEffect(() => {
+    if (primaryData) {
+      const screens: string[] = new Array();
+      primaryData?.data.forEach(row => {
+        if (row.transaction) {
+          screens.push(String(row.transaction));
+        }
+      });
+      setVisibleScreens(screens);
+    }
+  }, [primaryData]);
+
+  const tableSearchFilters = new MutableSearch(['transaction.op:ui.load']);
+  const derivedQuery = getTransactionSearchQuery(location, primaryEventView.query);
+
+  const combinedData = useMemo((): TableData | undefined => {
+    if (primaryData !== undefined && secondaryData !== undefined) {
+      const meta: MetaType = {};
+      meta.units = {
+        ...secondaryData.meta?.units,
+        ...primaryData.meta?.units,
+      };
+      meta.fields = {
+        ...secondaryData.meta?.fields,
+        ...primaryData.meta?.fields,
+      };
+
+      const data = primaryData.data.map(row => {
+        const matchingRow = secondaryData.data.find(
+          metricRow => metricRow.transaction === row.transaction
+        );
+        if (matchingRow) {
+          return {
+            ...matchingRow,
+            ...row,
+          };
+        }
+        return row;
+      });
+
+      return {
+        data: data,
+        meta: meta,
+      };
```

Comment:
I'm surprised this didn't auto-format but if the key and value are the same, you can just write it like this:

```suggestion
      return {
        data,
        meta,
      };
```